### DISCUSSION TITLE
Can't add card

### DISCUSSION THREAD

**Discussion by rpopisw** [mercadopago/sdk-js#45] [Q&A] [answered]
Hola compañeros, estoy tratando de agregar una tarjeta pero recibo el siguiente error 
```
{
	"message": "invalid parameters ",
	"error": "bad_request",
	"status": 400,
	"cause": [
		{
			"code": "118",
			"description": "the body must be a Json Object"
		}
	]
}
```
a pesar que estoy enviando un objeto Jsonn a la funcion de esta forma 

<img width="566" alt="Captura de Pantalla 2022-01-14 a la(s) 13 35 31" src="https://user-images.githubusercontent.com/69777661/149567436-e88f39e3-a9ba-460d-96d0-80c8610fc8f0.png">

he tratado de hacer lo mismo directo al api. pero obtengo el mismo resultado
<img width="1186" alt="Captura de Pantalla 2022-01-14 a la(s) 13 36 19" src="https://user-images.githubusercontent.com/69777661/149567540-b2240bef-797e-4199-ac4a-7988157914d8.png">
 como pueden ver en la imagen estoy que le paso un objeto json al api. pero la respuesta sigue siendo que el body no es un json, a que se debe este problema porfavor

**Comment by lucmantovani** [comment, marked as answer]
Hola, @rpopisw!

Posiblemente el error ocurrió porque la api espera el payload en el siguiente formato:
```json
{
  "token": "CARD_TOKEN",
  "issuer": {
    "id": "24"
  },
  "payment_method_id": "master"
}
```

Pero también puedes intentar enviar solo el token.

Por favor avísame si logras agregar la tarjeta!

**Reply by rpopisw** [reply]
Hola @lucmantovani gracias por la respuesta 

acabo de probar mandando el payload de esta forma 

<img width="1192" alt="Captura de Pantalla 2022-01-17 a la(s) 10 07 19" src="https://user-images.githubusercontent.com/69777661/149793969-17ca2331-0848-42df-9d41-dd4269658a0a.png">

pero como pueden ver me sale un error 
```
{
"code": "127",
"description": "invalid parameter. Cannot resolve the payment method of card, check the payment_method_id and issuer_id"
}
```
y si prueba mandando solo el token. 
<img width="1190" alt="Captura de Pantalla 2022-01-17 a la(s) 10 10 13" src="https://user-images.githubusercontent.com/69777661/149794493-f6e16d4f-a28a-4f56-b570-e08f72ded76f.png">

me sale el mismo error.

**Reply by lucmantovani** [reply]
Hola, @rpopisw!

Debido a que no funcionó solo con el token, esta tarjeta necesita la información de issuer.id y payment_method_id.

Por favor, compruebe el valor de estos dos parámetros.
Deben coincidir con la tarjeta utilizada al crear el token.

En su caso, creo que el error está en issuer.id, se la tarjeta es Visa.
Recomiendo usar los valores devueltos por el JS SDK.

**Reply by rpopisw** [reply]
Gracias @lucmantovani,  

estos son los valores de la tarjeta que obtengo desde el front usando el sdk
<img width="414" alt="Captura de Pantalla 2022-01-17 a la(s) 10 49 18" src="https://user-images.githubusercontent.com/69777661/149800949-4758cdb3-c4b2-48c6-b38f-168e9a3f218a.png">

pero obtengo el mismo error del body

<img width="1181" alt="Captura de Pantalla 2022-01-17 a la(s) 10 50 50" src="https://user-images.githubusercontent.com/69777661/149801165-881a7715-ffd4-4814-a29c-4280f51008ae.png">

 **cabe señalar que estoy usando tarjetas de prueba y tambien estoy usando las credenciales de pruebas**

**Reply by lucmantovani** [reply]
En este último caso, si envía el siguiente payload:
```json
{
  "token": "TOKEN",
  "issuer": {
    "id": "12551"
  },
  "payment_method_id": "visa"
}
```

 recibe el error `"invalid parameter. Cannot resolve the payment method of card, check the payment_method_id and issuer_id"`?

**Reply by rpopisw** [reply]
gracias por el apoyo @lucmantovani , si da el mismo error. Revisando desde el front el encargado de generar el token y pude notar que no enviaba toda la informacion a la libreria en el front, pero aun asi se generaba un token como si estubiera correcto la operacion. me asegure que se mande todos los datos desde el front para generar el token segun la documentacion y ahora si funciona correctamente.

**Reply by lucmantovani** [reply]
¡Me alegro de que lo hayas hecho y gracias por avisarme!

Cerrare esta discussion ahora.
Si tiene más problemas, puede contactarnos nuevamente.

Recordando que en los casos en que el error no esté específicamente en los SDKs, puedes solicitar soporte [aquí](https://www.mercadopago.com.ar/developers/es/support/center/tickets)
